homeassistant/components/shopping_list/__init__.py

```diff
@@ -217,6 +246,22 @@ async def async_add(self, name, context=None):
         )
         return item
 
+    async def async_remove(self, item_id, context=None):
+        """Remove a shopping list item."""
+        item = next((itm for itm in self.items if itm["id"] == item_id), None)
+
+        if item is None:
+            raise KeyError
```

Comment:
It would be safer to raise a custom specific exception, I think. There are other dict look ups in the path until we catch the `KeyError` that we don't expect to fail.